sklearn/metrics/classification.py

```diff
@@ -240,6 +240,8 @@ def confusion_matrix(y_true, y_pred, labels=None, sample_weight=None):
         labels = unique_labels(y_true, y_pred)
     else:
         labels = np.asarray(labels)
+        if np.all([l not in y_true for l in labels]):
```

Comment:
Ah. Of course yours is faster in the (usual) case that `np.all` will return False. When `0` is found in `x`, it will shortcut to returning False. As long as this is not painfully slow where `np.all` is to return `True` (i.e. `labels = [6,7,8,9,10]`), I'm okay with the current approach.